hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractParentQueue.java

```diff
@@ -1347,6 +1348,18 @@ public List<CSQueue> getChildQueues() {
 
   }
 
+  @Override
+  public List<CSQueue> getChildQueuesByTryLock() {
+    try {
+      while (!readLock.tryLock()){
+        LockSupport.parkNanos(10000);
```

Comment:
Is it better for us to add a configuration? This configuration is used to configure `10000`

Reply:
I don't know how useful is to configure this value, the only reason to increase it if the loop is too busy with the 10 thousand nanos (=10 microseconds).

https://hazelcast.com/blog/locksupport-parknanos-under-the-hood-and-the-curious-case-of-parking/

Based on this, the granularity should be around 50 microseconds anyway.

Reply:
I have no other comments. we need to fix checkstyle, then we can merge this pr.

Reply:
Ok, thanks. Fixed the checkstyle issues.